### ISSUE TITLE
GA: Organize key permissions for GCP KMS keys

### ISSUE THREAD

**Issue by asraa** [sigstore/root-signing#349]
**Description**

1. Remove test & staging keys: contingent on https://github.com/sigstore/root-signing/issues/312
2. Move all online signing keys (the sigstore-root) to two separated GCP projects: one for delegations and one for snapshot and timestamp
3. Create service accounts in each of these projects that are configured with workload identity to access these keys for signing.
4. Create a final service account that only has write access to sync metadata to project-rekor's GCS bucket: https://github.com/sigstore/root-signing/issues/75

Part of https://github.com/sigstore/public-good-instance/issues/376
@var-sdk

**Comment by asraa** [comment]
1. Removed sigstore-root test key (no longer needed) `projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/test` and staging `projects/projectsigstore-staging/locations/global/keyRings/root-keyring/cryptoKeys/staging-test`


@priyawadhwa

**Comment by asraa** [comment]
2. For online delegations keys: i need a sigstore group including keyholders and orchestrators (me, and others who can execute key signing ceremonies). AI: Create this group.
3. For snapshot/timestamp: only the workload identity people need access. Does that violate https://github.com/sigstore/sigstore/issues/467?

**Comment by asraa** [comment]
Discussion with Simon:
Concern here is that people with access with public-good-instance could do something with the service account that has access to the root signing keys.

AIs after discussing with Simon:
1. Create a GCP project hosting the online keys owned by tuf-team. These keys are: `projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/rekor_delegation`, `projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/revocation`, `projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/snapshot`, `projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/timestamp`, `projects/project-rekor/locations/global/keyRings/sigstore-root/cryptoKeys/staging-projects` (FYI `test` should have been deleted, but it looks like it was not)
3. Create 1 service accounts there: 
    4. `sigstore-root-sign@project-xyz.iam.gserviceaccount.com`. This will be authorized for the online GCP keys.
7. And add workflow identity in that project: It should give repository sigstore/root-signing authorization access. There are designated workflows for signing, if a principal can be scoped to a workflow, even better. 

In the current project-rekor,
 1.. Create a service account `sigstore-root-sync@project-rekor.iam.gserviceaccount.com`. This will be authorized for GCS bucket write operations. The workflow identity pool should be configured for sigstore/root-signing, and if a workflow is possible, we can designate it to the sync workflow.
2. Deprecate `sigstore-root-signing@project-rekor.iam.gserviceaccount.com` when this migration is compelte.


@var-sdk

**Comment by Hayden-IO** [comment]
That sounds like a great approach. 

Do we need a dedicated GCP project to host the keys? Or just better fine grained permissions?

**Comment by var-sdk** [comment]
Finer grained permissions for use are possible. They would still be accessible by project-rekor (& public-good-instance) admins.
